jwst/datamodels/model_base.py

```diff
@@ -140,7 +141,7 @@ def __init__(self, init=None, schema=None, extensions=None,
 
             shape = init
             is_shape = True
-            asdf = self.open_asdf(init=None, extensions=self._extensions,
+            asdffile = self.open_asdf(init=None, extensions=self._extensions,
                                   **kwargs)
 
         elif isinstance(init, DataModel):
```

Comment:
I'm not pleased that `asdffile` goes undefined in this block.  Yet our tests pass.  I still feel it is a bug.

Reply:
Maybe we should set `asdffile = None` above just for consistency/sanity? FWIW this seems like something that would be picked up by a code style check.

Reply:
That works.

Your code style checker must be better than mine!  😂

Reply:
Er, maybe I meant a linter more correctly. But I'm not sure if anyone is using `pylint` around here?

Reply:
I now see that the `return` means the lower code below does not get run.  🤦‍♂️ So yes, just style, not a bug.

Reply:
We've been using `flake8`, but that doesn't catch this.